dynamic_env_pytest_tests/tests/services/s3_gate/test_s3_gate.py

```diff
@@ -0,0 +1,559 @@
+            uploads = s3_gate_object.list_multipart_uploads_s3(self.s3_client, bucket)
+            assert not uploads, f"Expected there is no uploads in bucket {bucket}"
+
+        with allure.step("Create new multipart upload and upload several parts"):
+            upload_id = s3_gate_object.create_multipart_upload_s3(
+                self.s3_client, bucket, object_key
+            )
+            for part_id, file_path in enumerate(part_files, start=1):
+                etag = s3_gate_object.upload_part_s3(
+                    self.s3_client, bucket, object_key, upload_id, part_id, file_path
+                )
+                parts.append((part_id, etag))
+
+        with allure.step("Check all parts are visible in bucket"):
+            got_parts = s3_gate_object.list_parts_s3(self.s3_client, bucket, object_key, upload_id)
+            assert len(got_parts) == len(
+                part_files
+            ), f"Expected {parts_count} parts, got\n{got_parts}"
+
+        s3_gate_object.complete_multipart_upload_s3(
+            self.s3_client, bucket, object_key, upload_id, parts
+        )
+
+        uploads = s3_gate_object.list_multipart_uploads_s3(self.s3_client, bucket)
+        assert not uploads, f"Expected there is no uploads in bucket {bucket}"
+
+        with allure.step("Check we can get whole object from bucket"):
+            got_object = s3_gate_object.get_object_s3(self.s3_client, bucket, object_key)
+            assert get_file_hash(got_object) == get_file_hash(file_name_large)
+
+        self.check_object_attributes(bucket, object_key, parts_count)
+
+    @allure.title("Test S3 Bucket tagging API")
+    def test_s3_api_bucket_tagging(self, bucket):
+        """
+        Test checks S3 Bucket tagging API (Put tag/Get tag).
+        """
+        key_value_pair = [("some-key", "some-value"), ("some-key-2", "some-value-2")]
+
+        s3_gate_bucket.put_bucket_tagging(self.s3_client, bucket, key_value_pair)
+        check_tags_by_bucket(self.s3_client, bucket, key_value_pair)
+
+        s3_gate_bucket.delete_bucket_tagging(self.s3_client, bucket)
+        check_tags_by_bucket(self.s3_client, bucket, [])
+
+    @allure.title("Test S3 Object tagging API")
+    def test_s3_api_object_tagging(self, bucket, simple_object_size):
+        """
+        Test checks S3 Object tagging API (Put tag/Get tag/Update tag).
+        """
+        key_value_pair_bucket = [("some-key", "some-value"), ("some-key-2", "some-value-2")]
+        key_value_pair_obj = [
+            ("some-key-obj", "some-value-obj"),
+            ("some-key--obj2", "some-value--obj2"),
+        ]
+        key_value_pair_obj_new = [("some-key-obj-new", "some-value-obj-new")]
+        file_name_simple = generate_file(simple_object_size)
+        obj_key = self.object_key_from_file_path(file_name_simple)
+
+        s3_gate_bucket.put_bucket_tagging(self.s3_client, bucket, key_value_pair_bucket)
```

Comment:
are we obliged by S3 protocol to `PutBucketTagging` before `PutObjectTagging`?

Reply:
> are we obliged by S3 protocol to `PutBucketTagging` before `PutObjectTagging`?

As I understand from the documentation - no, we don't have to use `PutBucketTagging` before `PutObjectTagging`.

I tested it - you are right, the tests run the same way.

See PR:
https://github.com/nspcc-dev/neofs-testcases/pull/705